torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -699,7 +706,17 @@ struct NCCLTraceBuffer {
       };
 
       dict.insert(input_sizes_key, read_sizes(e.input_dims_));
+      std::vector<std::string> input_dtypes_strs;
```

Comment:
```suggestion
      std::vector<std::string> input_dtypes_strs;
      input_dtypes_str.reserve(e.inputs_dtypes.size());
```
We should be reserving the size of the vectors if we can.